artemis-commons/src/main/java/org/apache/activemq/artemis/utils/ActiveMQThreadPoolExecutor.java

```diff
@@ -46,32 +45,70 @@
    private static class ThreadPoolQueue extends LinkedBlockingQueue<Runnable> {
 
       private ActiveMQThreadPoolExecutor executor = null;
+      private int idleThreads = 0;
 
       public void setExecutor(ActiveMQThreadPoolExecutor executor) {
          this.executor = executor;
       }
 
       @Override
-      public boolean offer(Runnable runnable) {
+      public synchronized boolean offer(Runnable runnable) {
```

Comment:
@bgutjahr @clebertsuconic Today I'll take a look at the code and will do some test on it :)
@bgutjahr Do you've tested it against the original with some benchmark tool like [JMH](http://openjdk.java.net/projects/code-tools/jmh/)?